src/NuGet.Core/NuGet.Frameworks/FrameworkNameProvider.cs

```diff
@@ -805,7 +810,12 @@ private void AddPortableOptionalFrameworks(IEnumerable<KeyValuePair<int, NuGetFr
         // Ordered lists of framework identifiers
         public void AddFrameworkPrecedenceMappings(IEnumerable<string> mappings)
         {
-            _frameworkPrecedence.AddRange(mappings);
+            int index = 0;
```

Comment:
There could be multiple mappings that are passed in. The index will be reset between each one. This wasn't fully working before either, but now it looks like it could hit more conflicts.